### ISSUE TITLE
Build error for `unimplemented()` with tuple parameters

### ISSUE THREAD

**Issue by oskarek** [pointfreeco/swift-issue-reporting#120]
This causes a build error on the current version (1.2.5), and on every 1.2.x version:

```swift
let _: ((String, Int)) -> String = unimplemented("")
```

without any helpful compiler error message. The same thing works without issue on version 1.1.2.

The problem seems to occur for functions with tuple parameters.

**Comment by mbrandonw** [comment]
Yes, unfortunately this looks like a compiler bug. Unfortunately it crashes on Xcode 16.0 beta 6 and Xcode 16.1 beta. I have filed a bug report [here](https://github.com/swiftlang/swift/issues/76010), and we will just have to wait until that is fixed. In the meantime you may want to bundle that tuple into a proper struct, or just implement an unimplemented closure by hand.

I will going to convert this to a discussion for now since I don't think there is much we can do, but if you find a workaround please let us know!
